adapters/transformers/typescript.js

```diff
@@ -15,6 +15,15 @@ var defaultCompileOptions = _.extend(Typescript.getDefaultCompilerOptions(), {
 module.exports = {
   matches: /\.ts$/,
   provides: ".js",
+  providesIndirect: function(pathname, files) {
```

Comment:
Very interesting approach. Likely applicable for your experiments w/ `webpack`.

Can you think of other uses that might need additional metadata to make a decision? I wonder if `function(requestPath, files)` is the right signature.